### DISCUSSION TITLE
why KubernetesPodOperator not add labels used to track pod when set full_pod_spec or pod_template_file?

### DISCUSSION THREAD

**Discussion by camper42** [apache/airflow#13604] [Q&A]
https://github.com/apache/airflow/blob/master/airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py#L472
`KubernetesPodOperator.create_new_pod_for_operator` not add labels to pod when `full_pod_spec` or `pod_template_file` set.

https://github.com/apache/airflow/blob/master/airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py#L317-L323
but these labels are used to list pods to find overlap.

So, when set `full_pod_spec` or `pod_template_file`, airflow will always launch new pod, this can be a problem.

---

By design or mistake? Or do I miss something?

---

we use `full_pod_spec` to set container working_dir and host_ipc

host ipc is needed for some pytorch task
https://github.com/pytorch/pytorch/issues/2926#issuecomment-364099144

@mik-laj @dimberman

**Comment by dimberman** [comment]
Hi @camper42 I'm not sure I understand the problem. The pod_template_spec is supposed to be a template that you can override values for. Are you saying there is overlap if two pods launch wtih the same template?

**Reply by camper42** [reply]
yes, suppose I have an template without custom labels.

https://github.com/apache/airflow/blob/master/airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py#L471
`self.pod_template_file` is not `None`, `airflow_version` and `kubernetes_pod_operator` label won't add.

so, a pod without any labels created.

https://github.com/apache/airflow/blob/master/airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py#L316-L323
as comment says, it use combination of labels to uniquely identify a running pod, and our pod don't have any label.

**Comment by mastryukov1990** [comment]
did you solve problem?
